apps/pixeloven/site/docs/recommendations/react-components.mdx

```diff
@@ -75,4 +75,53 @@ ADD LINK TO DOCUMENTATION ON ROUTES
 
 ## Component composition
 
-ADD DOCUMENTATION ON COMPONENT COMPOSITION
+When it comes to extracting components, there are two important factors to consider:
+1. complexity of code (e.g. conditional rendering and other business logic)
+2. reusability of code
+
+We use the atomic design as our philosophy for creating reusable React components, and create a new storybook entry for each new atomic design component. However, sometimes an atomic design component itself might include complex business logic for rendering, or code that is reused only within that component, and extracting some of its pieces into their own components would help improve readability and maintainability of the code. If you see that these pieces need to be extracted into their own component, yet they do not warrant creating a new atomic design component in storybook, then simply create new React component files directly within the same storybook folder.
+
+In the example below, `CampaignMetadataIndicator`, is a molecule storybook component and it requires complex business for rendering several indicators. The `VelocityIndicator` and `ProximityIndicator` in the end both render the same molecule component, however they use widely different business logic and copy. Extracting these implementations into two separate components lets us clearly see the logic that the parent, `CampaignMetadataIndicator`, uses to decide which indicator to render. `VelocityIndicator` and `ProximityIndicator`, in turn, are responsible for logic specific to them. Since they are not being used anywhere else on the site, they do not warrant creation of a new storybook component (e.g. atom, molecule, ...), so they are kept within the same directory as their parent, `molecules/CampaignMetadataIndicator`.
```

Comment:
molecule storybook component => Molecule component

I think calling them Storybook components is confusing, storybook is really just an interface to preview, test, and document the components.

Reply:
Updated - [commit](https://github.com/pixeloven/pixeloven/pull/288/commits/5dcb3e77c76ddcf97a87c4bddd880b8278a6829c)